### PR TITLE
Ensure first control point starts at 0 in most common beat length calculation

### PR DESCRIPTION
**Pull request by smoogipoo**
This function was made to replicate the functionality of osu-stable's `Beatmap.BeatsPerMinuteRange.Z`. osu-stable forces the first control point to start at 0 for this calculation.

Note that this change also affects osu-difficulty-calculator, so the values currently displayed on osu-web sometimes differ from those displayed in osu-stable.

For example: https://osu.ppy.sh/beatmapsets/1592995#mania/3253520
In this beatmap the first control point has t=274, so the control point is extended by +274ms which makes it the most common value.
```
Current osu-lazer: 279.8 BPM
osu-web          : 279.8 BPM (via osu-difficulty-calculator)
osu-stable       : 215 BPM
This PR          : 215 BPM
```

### CODE REVIEW

**Comment by smoogipoo** [comment]
Unrelated to this PR, after merging I'm going to rewrite this method to follow osu-stable's code since it's quite hard to follow and inefficient:
```
|     Method | Count |        Mean |     Error |    StdDev |  Gen 0 | Allocated |
|----------- |------ |------------:|----------:|----------:|-------:|----------:|
| TestSimple |     1 |    158.4 ns |   1.13 ns |   1.00 ns | 0.0029 |     256 B |
|   TestLinq |     1 |    572.8 ns |  11.34 ns |  10.61 ns | 0.0086 |     792 B |
| TestSimple |    10 |    694.7 ns |   3.85 ns |   3.22 ns | 0.0114 |   1,032 B |
|   TestLinq |    10 |  2,017.2 ns |  20.12 ns |  18.82 ns | 0.0229 |   2,144 B |
| TestSimple |   100 |  3,764.5 ns |  19.56 ns |  17.34 ns | 0.0534 |   4,664 B |
|   TestLinq |   100 |  9,459.6 ns | 182.52 ns | 230.83 ns | 0.0916 |   8,920 B |
| TestSimple |  1000 |  5,656.1 ns |  50.55 ns |  47.28 ns | 0.0534 |   4,664 B |
|   TestLinq |  1000 | 15,597.8 ns |  98.29 ns |  87.13 ns | 0.1526 |  14,080 B |
| TestSimple | 10000 |  5,420.5 ns |  63.69 ns |  53.19 ns | 0.0534 |   4,664 B |
|   TestLinq | 10000 | 14,621.7 ns | 169.31 ns | 158.38 ns | 0.1373 |  12,560 B |
```

**Comment by peppy** [comment]
Hmm, trying to discern if this actually makes more or less sense. I think the way we had it would choose better values if the consideration is "the most common BPM used during gameplay", correct?

If we're going to go with the stable compatibility way, it's probably good to have an inline comment explaining why we're doing this.

**Comment by smoogipoo** [comment]
It's probably best to preserve compatibility across all systems for now? Mania's scroll speed, the display in song select, osu-web's beatmap listing?

Will add an inline comment.

**Review by peppy — APPROVED** [review]
